### PR TITLE
FS2 Loss included padded frames

### PR DESCRIPTION
**Pull request by roedoejet**
### PR Goal? 

Previously, the loss was calculated by multiplying against the mask, but this would still include the padded frames in MSE loss, unnecessarily. Here we switch to masked select instead to only calculate loss on non-padded inputs/outputs.

### Fixes? 

Issue raised by personal communication with @korin-richmond

### Feedback sought? 

sanity

### Priority? 

medium

### Tests added? 

yes

### How to test? 

I've included tests. you should be able to train a new model or finetune existing models without issues

### Confidence? 

high

### Version change? 

patch

### Related PRs? 

https://github.com/EveryVoiceTTS/FastSpeech2_lightning/pull/166

### CODE REVIEW

**Comment by joanise** [comment]
Oops, you said here you added tests, but they're not there. Did you forget to commit them?

**Review by joanise — APPROVED** [review]
lgtm
